picard/ui/metadatabox/tagdiff.py

```diff
@@ -342,3 +345,28 @@ def update_tag_names(self, changes_first=False, top_tags=None):
             self.tag_names = [
                 tag for tag in tag_names if
                 self.status[tag] != TagStatus.EMPTY]
+
+    def to_json(self):
+        result = {}
+        for tag in self.tag_names:
+            result[tag] = {}
+            if tag in self.old:
+                result[tag]['old'] = self.old[tag]
+            if tag in self.new:
+                result[tag]['new'] = self.new[tag]
+
+        return json.dumps(result)
+
+    def to_tsv(self):
+        COL_SEP = "\t"
+
+        f = io.StringIO()
+        writer = csv.writer(f, delimiter=COL_SEP)
+
+        for tag in self.tag_names:
+            writer.writerow([
+                tag,
+                MULTI_VALUED_JOINER.join(self.old[tag]),
+                MULTI_VALUED_JOINER.join(self.new[tag])])
+
+        return f.getvalue().rstrip()
```

Comment:
rstrip() can be a problem if last field contains whitespaces, so better use [`removesuffix()`](https://docs.python.org/3/library/stdtypes.html#str.removesuffix)